Document how to use with Cert-Manager (#475)
stackabletech/secret-operator: docs/modules/secret-operator/pages/cert-manager.adoc

https://cert-manager.io/[Cert-Manager] is a common tool to manage certificates in Kubernetes, especially when backed by an external
Certificate Authority (CA) such as https://letsencrypt.org/[Let\'s Encrypt].

The Stackable Secret Operator does not currently support managing Cert-Manager certificates directly, but it can be configured to consume certificates generated by it.

[#caveats]
== Caveats

Cert-Manager is designed to manage relatively long-lived certificates that are stored in Kubernetes Secrets. By contrast,
the Stackable Secret Operator is designed to generate temporary short-lived certificates.

This has a couple of repercussions:

- Longer-lived certificates mean that a leaked certificate has potential to be abused for longer.
- Application teams may have access to read Secrets in their respective applications' Namespaces.

Where possible, we recommend using the xref:secretclass.adoc#backend-autotls[`autoTls` backend] instead.

[#issuer]
== Configuring Cert-Manager

NOTE: We recommend using the xref:secretclass.adoc#backend-autotls[`autoTls` backend] instead for self-signed PKIs. We use Cert-Manager's CA issuer here to show the broader concepts.

> To do this, we'll first need to teach Cert-Manager how we want it to create our certificates.
> 
> In a production setup this will likely use an external CA such as ACME or OpenBao/Vault. However, to make this guide self-contained we will let Cert-Manager create
> a self-signed CA certificate instead.

Reviewer: As per the [Documentation style guide][doc-style], we should write in first person.

```suggestion
To do this, you will first need to teach Cert-Manager how to create your certificates.

In a production setup this will likely use an external CA such as ACME or OpenBao/Vault. However, to make this guide self-contained, Cert-Manager will create
a self-signed CA certificate instead.
```

[doc-style]: https://docs.stackable.tech/home/stable/contributor/docs-style-guide#_tone_and_writing_style_google_developer_documentation_style_guide

[source,yaml]
----
include::example$cert-manager/issuer.yaml[]
----
<1> This is the Issuer that our created certificates will reference later

[#secretclass]
== Creating a SecretClass

The Stackable Secret Operator needs to know how to find the certificates created by Cert-Manager. We do this by creating
a xref:secretclass.adoc[] using the xref:secretclass.adoc#backend-k8ssearch[`k8sSearch` backend], which can find arbitrary
Kubernetes Secret objects that have the correct labels.

[source,yaml]
----
include::example$cert-manager/secretclass.yaml[]
----
<1> Both certificates and Pods will reference this name, to ensure that the correct certificates are found
<2> This informs the Secret Operator that certificates will be found in the same namespace as the Pod using it

[#certificate]
== Requesting a certificate

> We can now use Cert-Manager to provision our first certificate. We use labels to inform the Stackable Secret Operator
> about which xref:scope.adoc[scopes] the certificate fulfils. Which scopes must be provisioned is going to depend
> on the applicable workload. This guide assumes the xref:scope.adoc#service[service] and xref:scope.adoc#node[node] scopes.

Reviewer: ```suggestion
You can now use Cert-Manager to provision your first certificate. Use labels to inform the Stackable Secret Operator
about which xref:scope.adoc[scopes] the certificate fulfills. Which scopes must be provisioned is going to depend
on the design of the workload. This guide assumes the xref:scope.adoc#service[service] and xref:scope.adoc#node[node] scopes.
```
